src/main/java/fi/aalto/cs/intellij/actions/RequiredPluginsCheckerAction.java

```diff
@@ -0,0 +1,211 @@
+package fi.aalto.cs.intellij.actions;
+
+import static java.util.stream.Collectors.toMap;
+
+import com.intellij.ide.plugins.IdeaPluginDescriptor;
+import com.intellij.ide.plugins.PluginManager;
+import com.intellij.ide.plugins.PluginManagerConfigurable;
+import com.intellij.ide.plugins.RepositoryHelper;
+import com.intellij.ide.plugins.newui.BgProgressIndicator;
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationAction;
+import com.intellij.notification.NotificationType;
+import com.intellij.notification.Notifications;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.extensions.PluginId;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.startup.StartupActivity;
+import com.intellij.openapi.updateSettings.impl.PluginDownloader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.StringJoiner;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+//TODO THINK HOW TO TEST & REFACTOR THIS
+public class RequiredPluginsCheckerAction implements StartupActivity {
+
+  private static final Logger logger = LoggerFactory.getLogger(RequiredPluginsCheckerAction.class);
```

Comment:
**logging:** default logging (slf4j) implementation used within the IJ framework